src/Xamarin.Android.Build.Tasks/Tasks/FixupNuGetReferences.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Framework;
+
+namespace Xamarin.Android.Tasks
+{
+	/// <summary>
+	/// This task contains *temporary* workarounds for NuGet in .NET 5.
+	/// </summary>
+	public class FixupNuGetReferences : AndroidTask
+	{
+		public override string TaskPrefix => "FNR";
+
+		[Required]
+		public string [] PackageTargetFallback { get; set; }
+
+		public ITaskItem [] CopyLocalItems { get; set; }
+
+		[Output]
+		public string [] AssembliesToAdd { get; set; }
+
+		[Output]
+		public ITaskItem [] AssembliesToRemove { get; set; }
+
+		public override bool RunTask ()
+		{
+			if (CopyLocalItems == null || CopyLocalItems.Length == 0)
+				return true;
+
+			var assembliesToAdd     = new Dictionary<string, string> ();
+			var assembliesToRemove  = new List<ITaskItem> ();
+			var fallbackDirectories = new HashSet<string> ();
+
+			foreach (var item in CopyLocalItems) {
+				var directory = Path.GetDirectoryName (item.ItemSpec);
+				var directoryName = Path.GetFileName (directory);
+				Log.LogDebugMessage ($"{directoryName} -> {item.ItemSpec}");
+				if (directoryName == "netstandard2.0") {
+					var parent = Directory.GetParent (directory);
+					foreach (var nugetDirectory in parent.EnumerateDirectories ()) {
+						var name = Path.GetFileName (nugetDirectory.Name);
+						foreach (var fallback in PackageTargetFallback) {
+							if (!string.Equals (name, fallback, StringComparison.OrdinalIgnoreCase))
+								continue;
+							var fallbackDirectory = Path.Combine (parent.FullName, fallback);
+							fallbackDirectories.Add (fallbackDirectory);
+
+							// Remove the netstandard assembly, if there is a platform-specific one
+							var path = Path.Combine (fallbackDirectory, Path.GetFileName (item.ItemSpec));
+							if (File.Exists (path)) {
+								Log.LogDebugMessage ($"Removing: {item.ItemSpec}");
+								assembliesToRemove.Add (item);
```

Comment:
Should we set a flag so we break out of the `EnumerateDirectories` loop if we find a match? reading this code it looks like we will continue to enumerate directories

Reply:
The latest Forms NuGet package has multiple `MonoAndroid` directories:
```
> 7z l 'C:\Users\jopepper\Downloads\xamarin.forms.4.6.0.726.nupkg' | grep Xamarin.Forms.Platform.dll
2020-04-28 02:01:24 .....       138880        45480  lib\MonoAndroid10.0\Xamarin.Forms.Platform.dll
2020-04-28 02:01:30 .....       127104        43270  lib\MonoAndroid90\Xamarin.Forms.Platform.dll
2020-04-28 02:01:34 .....        13952         7171  lib\netstandard1.0\Xamarin.Forms.Platform.dll
2020-04-28 02:01:26 .....        13952         7164  lib\netstandard2.0\Xamarin.Forms.Platform.dll
2020-04-28 02:01:30 .....        19584         7924  lib\tizen40\Xamarin.Forms.Platform.dll
2020-04-28 02:01:36 .....        13952         7173  lib\uap10.0.14393\Xamarin.Forms.Platform.dll
2020-04-28 02:01:28 .....        13952         7164  lib\uap10.0.16299\Xamarin.Forms.Platform.dll
2020-04-28 02:01:26 .....        18560         7560  lib\Xamarin.iOS10\Xamarin.Forms.Platform.dll
2020-04-28 02:01:34 .....        13952         7166  lib\Xamarin.Mac\Xamarin.Forms.Platform.dll
```
I think we have to continue the loop in case the `MonoAndroid90` is the one that needs removed...

This is _hopefully_ going to be a hack we can remove later.